File: Scraping scripts/scraping.py
import sys
reload(sys)
sys.setdefaultencoding('utf-8')
import requests
import csv
from bs4 import BeautifulSoup
 
with open('data.csv', 'wb') as csvfile:
	filewriter = csv.writer(csvfile, delimiter=',',quotechar='|', quoting=csv.QUOTE_MINIMAL)
	filewriter.writerow(['Name', 'Address','Rate per sq ft','Price',' Area','Years',' Bathrooms','Floor'])
	url = ""	#Enter URL here
	for counter in range(2,100):
		url2 = url + str(counter)
		r =requests.get(url2)
		soup = BeautifulSoup(r.content,"lxml")

		g_data = soup.find_all("div",{"class" : "infoWrap"})
		for item in g_data:
			try: 
				a = item.contents[0].contents[0].text
			except:
				a = NaN
			try:		
				b = item.contents[0].contents[1].text.replace(',','')
			except:
				b = NaN
			try:		
				x = item.contents[1].contents[1].contents[0].contents[0].contents[1].text.encode('utf-8')
			except:	
				x = NaN
			try:	
				y = item.contents[1].contents[1].contents[0].contents[0].contents[2].text.encode('utf-8')
			except:
				y = NaN	
			pr = x+' '+y
			try:
				c = item.contents[1].contents[2].contents[0].text
			except:
				c = NaN
			try:	
				d = item.contents[2].contents[0].contents[0].text
			except:
				d = NaN	
			try:
				e = item.contents[2].contents[1].text
			except:
				e = NaN
			try:
				f = item.contents[2].contents[2].text
			except:
				f = NaN 
			try:		
				g = item.contents[1].contents[1].contents[1].text.encode('utf-8')
			except:
				g = NaN	
			filewriter.writerow((a,b,g,pr,c,d,e,f))

File: Scraping scripts/scraping.rb
require 'open-uri'
require 'nokogiri'
require 'csv'

a = []

price = []
address = []
area = []
persqft = []
room = []
bathroom = []
verify = []
floor = []
community = []
power = []
secure = []

100.times do |i|
	url = ""	#Enter URL here
	page = Nokogiri::HTML(open(url))

	page.css('div.listing_list div.listing_list__content div.listing__title div.result__details__price').each do |line|
		price << line.text.strip.split(/ \n /)
	end

	page.css('div.listing_list div.listing_list__content div.listing__title div.listing__title__text').each do |line|
		address << line.text.strip.split(/ \n /)
	end

	page.css('div.listing_list div.listing_list__content div.listing__keywords span.summary-item.size').each do |line|
		area << line.text.strip.split(/ \n /)
	end

	page.css('div.listing_list div.listing_list__content div.listing__keywords span.summary-item.size-per').each do |line|
		persqft << line.text.strip.split(/ \n /)
	end

	page.css('div.listing_list div.listing_list__content div.listing__keywords span.summary-item.rooms').each do |line|
		room << line.text.strip.split(/ \n /)
	end

	page.css('div.listing_list div.listing_list__content div.listing__keywords span.summary-item.bathrooms').each do |line|
		bathroom << line.text.strip.split(/ \n /)
	end

	page.css('div.listing_list div.listing_list__content div.listing__keywords span.summary-item.property_verified').each do |line|
		verify << line.text.strip.split(/ \n /)
	end

	page.css('div.listing_list div.listing_list__content div.listing__keywords span.summary-item.independent_floor').each do |line|
		floor << line.text.strip.split(/ \n /)
	end

	page.css('div.listing_list div.listing_list__content div.listing__keywords span.summary-item.gated').each do |line|
		community << line.text.strip.split(/ \n /)
	end

	page.css('div.listing_list div.listing_list__content div.listing__keywords span.summary-item.power_backup').each do |line|
		power << line.text.strip.split(/ \n /)
	end

	page.css('div.listing_list div.listing_list__content div.listing__keywords span.summary-item.security').each do |line|
		secure << line.text.strip.split(/ \n /)
	end
end

CSV.open("h.csv", "w") do |file|
	file << ["Price", "Address", "Area", "Per_Square_Price", "Room", "Bathroom", "Verified?", "Floor", "Community", "Power_Backup", "Security"]

	price.length.times do |i|
		file << [price[i], address[i], area[i], persqft[i], room[i], bathroom[i], verify[i], floor[i], community[i], power[i], secure[i]]
	end
end
